services/authentication/sign_up.rb

```diff
@@ -0,0 +1,33 @@
+class AuthenticationService
+  class SignUp < ApplicationService
+    def initialize(params)
+      @params = params
+    end
+
+    def call
+      if UserPolicy.new(params).user_info_valid?
+        if email_available?
+          User.create(email: params[:email], password: params[:password])
+          RegistrationMailer.new(email: params[:email]).send
+          :success
+        else
+          :email_unavailable
```

Comment:
This looks messy, think about the way to refactor this
A few notes

1. Move policy initialization to a separate method
2. Make use of guard clauses such as `return unless policy.valid?`
3. Mailer should send an email only and only if user has been created. 
`create` [from Rails doc] - object is returned whether the object was saved successfully to the database or not. 
Use `persisted?` or something similar to verify that user has been saved to DB

4. Returning symbols looks ugly AF. You've got validations for presence & uniqueness and errors hash after object creation therefore you don't need symbols at all. Let me know if you want me to evaluate on this point